src/modules/sq/process.cpp

```diff
@@ -33,76 +39,205 @@ bool SQModule::process(Dissolve &dissolve, ProcessPool &procPool)
-            continue;
-        }
+        Messenger::print("SQ: Unweighted partial S(Q) are up-to-date.\n");
+        return true;
+    }
 
-        // Transform g(r) into S(Q)
-        if (!calculateUnweightedSQ(procPool, cfg, unweightedgr, unweightedsq, qMin, qDelta, qMax, cfg->atomicDensity(),
-                                   windowFunction, qBroadening))
-            return false;
+    // Transform g(r) into S(Q)
+    if (!calculateUnweightedSQ(procPool, unweightedgr, unweightedsq, qMin, qDelta, qMax, rho, windowFunction, qBroadening))
+        return false;
 
-        // Set names of resources (Data1D) within the PartialSet
-        unweightedsq.setObjectTags(fmt::format("{}//{}//{}", cfg->niceName(), "SQ", "UnweightedSQ"));
-        unweightedsq.setFingerprint(fmt::format("{}", cfg->moduleData().version("UnweightedGR")));
+    // Include Bragg scattering?
+    if (includeBragg)
+    {
+        // Get associated BraggModule
+        const auto *braggModule = keywords_.retrieve<const BraggModule *>("SourceReflections", nullptr);
+        if (!braggModule)
+            return Messenger::error("A source Bragg module must be provided.\n");
+
+        // Check if reflection data is present
+        if (!dissolve.processingModuleData().contains("BraggReflections", braggModule->uniqueName()))
+            return Messenger::error("Bragg scattering requested to be included, but reflections from the module '{}' "
+                                    "could not be located.\n",
+                                    braggModule->uniqueName());
+        const auto &braggReflections = GenericListHelper<Array<BraggReflection>>::value(
+            dissolve.processingModuleData(), "BraggReflections", braggModule->uniqueName(), Array<BraggReflection>());
+        const auto nReflections = braggReflections.nItems();
+        const auto braggQMax = braggReflections.constAt(nReflections - 1).q();
+        Messenger::print("Found BraggReflections data for module '{}' (nReflections = {}, QMax = {} "
+                         "Angstroms**-1).\n",
+                         braggModule->uniqueName(), nReflections, braggQMax);
+
+        // Create a temporary array into which our broadened Bragg partials will be placed
+        bool created;
+        auto &braggPartials = GenericListHelper<Array2D<Data1D>>::realise(dissolve.processingModuleData(), "BraggPartials",
+                                                                          uniqueName_, GenericItem::NoFlag, &created);
+        if (created)
+        {
+            // Initialise the array
+            braggPartials.initialise(unweightedsq.nAtomTypes(), unweightedsq.nAtomTypes(), true);
+
+            for (int i = 0; i < unweightedsq.nAtomTypes(); ++i)
+            {
+                for (int j = i; j < unweightedsq.nAtomTypes(); ++j)
+                    braggPartials.at(i, j) = unweightedsq.partial(0, 0);
+            }
```

Comment:
```suggestion
            for_each_pair (0, unweightedsq.nAtomTypes(), [&](const int i, const int j)
            {
                    braggPartials.at(i, j) = unweightedsq.partial(0, 0);
            });
```

Might as well use your hard work